servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -220,9 +223,12 @@ public void onNext(final ServiceDiscovererEvent<ResolvedAddress> event) {
             private Host<ResolvedAddress, C> createHost(ResolvedAddress addr) {
                 Host<ResolvedAddress, C> host = new Host<>(addr);
                 if (!eagerConnectionShutdown) {
-                    host.onClose().afterFinally(() ->
-                            activeHostsUpdater.updateAndGet(RoundRobinLoadBalancer.this,
-                                    previousHosts -> removeFromHostsList(previousHosts, addr))
+                    host.onClose().afterFinally(() -> {
+                                usedHostsUpdater.updateAndGet(RoundRobinLoadBalancer.this,
+                                        previousHosts ->
+                                                listWithHostRemoved(previousHosts, current -> current == host));
+                                host.markClosed();
```

Comment:
IIUC, this callback can be invoked only if the host is already marked as closed, no need to mark it again.